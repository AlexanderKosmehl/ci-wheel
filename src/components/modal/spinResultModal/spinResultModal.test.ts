import {
  describe, expect, it, jest,
} from '@jest/globals';
import generateModal from './spinResultModal';

describe('generateModal', () => {
  const onCloseCallback = jest.fn(() => {});
  const onDeleteCallback = jest.fn(() => {});

  const modal = generateModal({
    label: 'Test',
    onClose: onCloseCallback,
    onDelete: onDeleteCallback,
  });
  const closeButton = modal.querySelector<HTMLButtonElement>('.closeButton');
  const deleteButton = modal.querySelector<HTMLButtonElement>('.deleteButton');

  it('generates component correctly', () => {
    expect(modal).toMatchSnapshot();
  });

  it('adds working onClose callback', () => {
    if (!closeButton) throw Error('No closeButton rendered!');

    closeButton.click();
    expect(onCloseCallback).toBeCalled();
  });

  it('adds working onDelete callback', () => {
    if (!deleteButton) throw Error('No deleteButton rendered!');

    deleteButton.click();
    expect(onDeleteCallback).toBeCalled();
  });
});
