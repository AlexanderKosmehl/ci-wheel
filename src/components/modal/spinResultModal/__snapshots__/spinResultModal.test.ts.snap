// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateModal generates component correctly 1`] = `
<div
  class="container"
>
  <div
    class="content"
  >
    <div
      class="header"
    >
      <h2
        class="headerText"
      >
        Ergebnis:
      </h2>
      <button
        class="button closeButton"
      >
        <img
          class="icon"
          src="test-file-stub"
        />
      </button>
    </div>
    <li
      class="container"
    >
      <span
        class="label"
      >
        Test
      </span>
      <button
        class="button deleteButton"
      >
        <img
          class="icon"
          src="test-file-stub"
        />
      </button>
    </li>
  </div>
</div>
`;
